pkg/proctree/types.go

```diff
@@ -0,0 +1,50 @@
+package proctree
+
+// ProcessInfo is the user facing representation of a process data at a specific time.
+type ProcessInfo struct {
+	Id              int
+	NsId            int
+	Ppid            int
+	NsPpid          int // TODO: is this redundant?
+	UserId          int
+	ContainerId     string
+	Cmd             []string
+	ExecutionBinary FileInfo
+	StartTime       int // First thread fork time. TODO: Can we use the mian thread start time instead?
+	ExecTime        int // Last execve call time
+	ExitTime        int
+	ExistingThreads []int
+	ChildrenIds     []int
```

Comment:
It may be confusing that children refer to child processes. I would be more explicit here - `childProcessIDs`